include/dali/core/mm/pool_resource.h

```diff
@@ -48,6 +56,25 @@ struct pool_options {
    * @remarks This option is ignored when `try_smaller_on_failure` is set to `false`.
    */
   bool return_to_upstream_on_failure = true;
+
+  /**
+   * @brief To what extent should `deallocate` synchronize before making the memory available
+   */
+  sync_scope sync = sync_scope::none;
+
+  /**
+   * @brief Enables deferred deallocation if the pool supports it (otherwise ignored)
+   */
+  bool enable_deferred_deallocation = false;
+
+  /**
+   * @brief Maximum number of outstanding deferred deallocations
+   *
+   * If there are more outstanding deferred deallocations than this number,
+   * the subsequent allocation blocks.
+   */
+  int max_outstanding_deallocations = 16;
```

Comment:
How you selected this number?

Reply:
Nothing special, really. In fact, it's quite conservative. I've tried many times to even observe bulk deallocation and only saw it once until I ran `stress - c7` on another console (8-thread machine). I considered specifying it in outstanding deallocated bytes (instead of number of allocations) - but then decided that small deallocations also constribute to fragmentation and, pessimistiaclly, this could lead to indefinite growth of the outstanding allocation queue.

Reply:
Understood